packages/data/src/namespace-store.js

```diff
@@ -100,10 +100,14 @@ function createReduxStore( reducer, key, registry ) {
  *                            public facing API. Selectors will get passed the
  *                            state as first argument.
  * @param {Object} store      The redux store to which the selectors should be mapped.
+ * @param {Object} registry   Registry reference.
+ *
  * @return {Object}           Selectors mapped to the redux store provided.
  */
-function mapSelectors( selectors, store ) {
-	const createStateSelector = ( selector ) => function runSelector() {
+function mapSelectors( selectors, store, registry ) {
+	const createStateSelector = ( registeredSelector ) => function runSelector() {
+		const selector = registeredSelector.isRegistrySelector ? registeredSelector( registry ) : registeredSelector;
```

Comment:
While it's fairly trivial, `runSelectors` is our hottest path in the application.

Is it possible to assign this once when `mapSelectors` is first called? The `registry` should stay constant, correct?

Reply:
Good catch, I originally wanted to implement this feature regardless of the store implementation (outside namespace-store) but it means the computation is done on each select so I moved it here to solve the issue but it seems like I didn't :P I just thought I did. I'll follow-up